src/main/java/uk/gov/digital/ho/hocs/domain/queue/migration/MigrationService.java

```diff
@@ -38,40 +37,146 @@ public class MigrationService {
     public MigrationService(MigrationCaseworkClient migrationCaseworkClient,
                             RequestData requestData,
                             ObjectMapper objectMapper,
-                            MessageLogService messageLogService) {
+                            MessageLogService messageLogService,
+                            DocumentClient documentClient) {
         this.migrationCaseworkClient = migrationCaseworkClient;
         this.requestData = requestData;
         this.objectMapper = objectMapper;
         this.messageLogService = messageLogService;
+        this.documentClient = documentClient;
     }
 
     public void createMigrationCase(MigrationData migrationCaseData, MigrationCaseTypeData migrationCaseTypeData) {
+        UUID caseId;
+        UUID stageId;
+        ResponseEntity<CreateMigrationCaseResponse> caseResponseEntity;
+        ResponseEntity correspondentsResponseEntity = null;
+        ResponseEntity caseAttachmentsResponseEntity = null;
+
+        // case
         try {
             var migrationRequest = composeMigrateCaseRequest(migrationCaseData, migrationCaseTypeData);
-            CreateCaseworkCaseResponse caseResponse = migrationCaseworkClient.migrateCase(migrationRequest);
-            messageLogService.updateCaseUuidAndStatus(requestData.getCorrelationId(), caseResponse.getUuid(), Status.CASE_CREATED);
+            caseResponseEntity = migrationCaseworkClient.migrateCase(migrationRequest);
+            CreateMigrationCaseResponse caseResponse = caseResponseEntity.getBody();
+            messageLogService.updateCaseUuidAndStatus(
+                    requestData.getCorrelationId(),
+                    caseResponse.getUuid(),
+                    Status.CASE_CREATED);
+            caseId = caseResponse.getUuid();
+            stageId = caseResponse.getStageId();
             log.info("Created migration case {}", caseResponse.getUuid());
         } catch (Exception e) {
             messageLogService.updateStatus(requestData.getCorrelationId(), Status.CASE_MIGRATION_FAILED);
-            throw new ApplicationExceptions.DocumentCreationException(e.getMessage(), LogEvent.CASE_MIGRATION_FAILURE);
+            throw new ApplicationExceptions.CaseCreationException(e.getMessage(), LogEvent.CASE_MIGRATION_FAILURE);
+        }
+
+        if (hasCreatedCase(caseResponseEntity)) {
```

Comment:
Spring should treat a non-200 as a exception (HttpClientErrorException, HttpServerErrorException). This would trigger the exception on line 70 making this redundant.

Reply:
All 3 checks removed.